roles/loki/files/config.j2

```diff
@@ -1,16 +1,24 @@
 auth_enabled: false
 
 server:
-  http_listen_address: 127.0.0.1
+  http_listen_address: 172.16.34.1
   http_listen_port: 3100
   grpc_listen_port: 9096
 
+compactor:
+  working_directory: {{ data_path }}/retention
+  shared_store: filesystem
+  compaction_interval: 10m
+  retention_enabled: true
+  retention_delete_delay: 2h
+  retention_delete_worker_count: 20
+
 common:
-  path_prefix: /data/loki
+  path_prefix: {{ data_path }}
   storage:
     filesystem:
-      chunks_directory: /data/loki/chunks
-      rules_directory: /data/loki/rules
+      chunks_directory: {{ data_path }}/chunks
+      rules_directory: {{ data_path }}/rules
```

Comment:
Great! However, Ansible variables are global (I know... :/) so I would suggest `loki_data_path`. I would also suggest defining `/var/lib/loki` in the defaults (ref https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html) for the role.